File: javaLoginCore/src/main/java/com/hamdan/forzenbook/java/login/core/domain/usecase/LoginValidationUseCaseImpl.java
package com.hamdan.forzenbook.java.login.core.domain.usecase;

import android.util.Log;

import com.hamdan.forzenbook.java.login.core.data.repository.LoginRepository;

import java.io.IOException;

public class LoginValidationUseCaseImpl implements LoginValidationUseCase {

    final private LoginRepository repository;

    public LoginValidationUseCaseImpl(LoginRepository repository) {
        this.repository = repository;
    }

    @Override
    public void invoke(String email) throws IOException {
        repository.requestValidation(email);
    }
}

File: loginCore/src/main/java/com/hamdan/forzenbook/login/core/domain/usecase/LoginValidationUseCaseImpl.kt
package com.hamdan.forzenbook.login.core.domain.usecase

import com.hamdan.forzenbook.login.core.data.repository.LoginRepository

class LoginValidationUseCaseImpl(val repository: LoginRepository) :
    LoginValidationUseCase {
    override suspend fun invoke(email: String) {
        repository.requestValidation(email)
    }
}
